modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java

```diff
@@ -158,6 +178,8 @@ private CompletableFuture<ClientChannel> initAsync(ClientConnectionMultiplexer c
                         log.debug("Connection established [remoteAddress=" + s.remoteAddress() + ']');
                     }
 
+                    new IgniteThread(timeoutWorker).start();
```

Comment:
Does this mean a thread per client? Isn't this too much?

Reply:
Do you think we can share the worker among clients started on one JVM?

Reply:
Why not :) The collection is concurrent anyway

Reply:
One timeout worker for all clients is enough. Maybe even one timeout worker per JVM for all Ignite needs, we can put it in the core module.

Reply:
I do not know how I can do it right now.
May I left TODO here?

Reply:
Do you have any questions, what exactly is not clear?
* Create a separate class
* Put a static concurrent map there
* Create one thread for timeout checks

Reply:
It is not clear where this thread has to be stopped.
A static variable does not seem like a good solution to me.
I believe it is better to do it on a separate ticket.

Reply:
Is a demon thread an option?

Reply:
Right now, we start/stop it in the general life cycle. I do not see a reason to make it a demon.